public/app/features/scenes/models/SceneComponentEditWrapper.tsx

```diff
@@ -0,0 +1,80 @@
+import { css } from '@emotion/css';
+import React, { CSSProperties } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { useStyles2 } from '@grafana/ui';
+
+import { SceneObjectBase } from './SceneObjectBase';
+import { SceneComponentProps } from './types';
+
+export function SceneComponentEditWrapper<T extends SceneObjectBase<any>>({
+  model,
+  isEditing,
+}: SceneComponentProps<T>) {
+  const Component = (model as any).constructor['Component'] ?? EmptyRenderer;
+  const inner = <Component model={model} isEditing={isEditing} />;
+
+  if (!isEditing) {
+    return inner;
+  }
+
+  return <SceneComponentEditingWrapper model={model}>{inner}</SceneComponentEditingWrapper>;
+}
+
+export function SceneComponentEditingWrapper<T extends SceneObjectBase<any>>({
+  model,
+  children,
+}: {
+  model: T;
+  children: React.ReactNode;
+}) {
+  const styles = useStyles2(getStyles);
+  const editor = model.getSceneEditor();
+  const { hoverObject, selectedObject } = editor.useState();
```

Comment:
this bit is pretty elegant I think. This component updates if the hover or selected object change, the change is either initiated from here from the editor side pane.  The inner SceneObject will not re-render.